protocols/grpc.go

```diff
@@ -92,18 +97,30 @@ func NewGrpcClientsIfConfigured(config *service.Config) ([]service.Client, error
 			clientID = config.GRPCProxy + " / " + serverURL
 		}
 
-		conn, err := grpc.Dial(
+		// we can throw away the CancelFunc because we use `grpc.WithBlock()` below
+		ctx, cancel := context.WithTimeout(context.Background(), config.DownstreamTimeout)
+		defer cancel()
```

Comment:
Super minor: am having a bit of trouble parsing this comment. It looks like you are deferring the call to cancel, so I'm not sure what's being thrown away.

Reply:
ahh good catch. you're correct the comment doesn't make sense, it applied to a previous commit, will remove.